storage/innobase/handler/ha_innodb.cc

```diff
@@ -16027,7 +16027,7 @@ ha_innobase::external_lock(
 	    && thd_sqlcom_can_generate_row_events(thd)) {
 		bool skip = false;
 #ifdef WITH_WSREP
-		skip = trx->is_wsrep() && !wsrep_thd_is_local(thd);
+		skip = wsrep_on(thd) && !wsrep_thd_is_local(thd);
 #endif /* WITH_WSREP */
```

Comment:
Can this be consolidated in `thd_sqlcom_can_generate_row_events(thd)` and the local variable `skip` be removed altogether? This code is its only caller.